### PR TITLE
[AUD-1464] Merge native and mobile navigation

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

First pass at merging native and web navigation into a single hook. Maintains strong typing for native navigation, and we can work on a similar strong typing for web too.

### CODE REVIEW

File: src/hooks/useNavigation.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+> = {
+  native: { screen: RouteName; params: ParamList[RouteName] }
+  web: {
+    route: string
+    fromPage?: string
+    fromNativeNotifications?: string
+  }
+}
+
+type NavigationType = NativeStackNavigationProp<BaseStackParamList>
```
Comment (sliptype) [inline src/hooks/useNavigation.ts:23]: `BaseStackParamList` might not be extensive enough for all our stacks, so we might need a type parameter for that at some point. But probably fine for now

File: src/components/track-tile/TrackTile.tsx [hunk 2]
```diff
@@ -107,14 +98,19 @@ const TrackTileComponent = ({
...
-    [pushRouteWeb, permalink, navigation, track_id]
+    [navigation, permalink, track_id]
   )
 
   const goToArtistPage = (e: GestureResponderEvent) => {
-    navigation.navigate('profile', { handle: user.handle })
+    navigation.navigate({
+      native: { screen: 'profile', params: { handle: user.handle } },
+      web: { route: user.handle }
+    })
```
Comment (sliptype) [inline src/components/track-tile/TrackTile.tsx:113]: I like this! My only thought is that there will be a fair amount of changes when we rip out web. But that should be fine

**Review by sliptype — APPROVED** [review]
Looks great! Happy that the long navigation type gets hidden away too

File: src/components/track-tile/TrackTile.tsx (hunk 2 above)
Comment (dylanjeffers) [reply src/components/track-tile/TrackTile.tsx:113]: yeah that is a really good call...

File: src/hooks/useNavigation.ts (hunk 1 above)
Comment (dylanjeffers) [reply src/hooks/useNavigation.ts:23]: word, can def try and extend as we use it in more complex places!

**Pull request merged**